optical_flow.rs

```diff
@@ -0,0 +1,127 @@
+use crate::spi::HypedSpi;
+use std::time::{Duration, Instant, SystemTime, UNIX_EPOCH};
+use byteorder::{LittleEndian, ReadBytesExt};
+use std::thread::sleep;
+use serde::{Serialize, Deserialize};
+use serde_json::json;
+
+
+// implement get_motion function first
+// add readByteFromRegister (?) function to HypedSPI trait which takes register address and gives data
+
+const REG_MOTION_BURST: u8 = 0x16;
+const TIMEOUT: Duration = Duration::from_secs(5);
+
+
+pub struct PMW3901<SPI> {
+    spi: SPI,
+
+}
+
+impl<SPI> PMW3901<SPI>
+where
+    SPI: HypedSpi,
+{
+
+    
+    pub fn new(spi: SPI) -> Self {
+        PMW3901 { spi }
+    }
+
+    // **ATTEMPTED PYTHON IMPLEMENTATION OF get_motion**
+    pub fn get_motion(&mut self) -> Result<(i16, i16), &'static str>{
```

Comment:
You'll want to define an error enum type like `OpticalFlowError` instead of returning a string here. Have a look at `TemperatureError` as an example (you'll probably want to have an enum variant for `SpiError` for example